futures/tests/io_lines.rs

```diff
@@ -0,0 +1,52 @@
+use futures::executor::block_on;
+use futures::future::Future;
+use futures::stream::{self, StreamExt, TryStreamExt};
+use futures::io::AsyncBufReadExt;
+use futures::task::Poll;
+use futures_test::io::AsyncReadTestExt;
+use futures_test::task::noop_context;
+use std::io::Cursor;
+use std::pin::Pin;
+
+#[test]
+fn lines() {
+    let buf = Cursor::new(&b"12\r"[..]);
+    let mut s = buf.lines();
+    assert_eq!(block_on(s.next()).unwrap().unwrap(), "12\r".to_string());
```

Comment:
nit: would you mind moving the `block_on(<arg>.next()).unwrap().unwrap()` into a function or macro so that it doesn't clutter the main test logic?